www/checktokenchallenges.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+    try{
+        $authStateId = $_REQUEST['StateId'];
+        $state = SimpleSAML_Auth_State::loadState($authStateId, 'privacyidea:privacyidea:init');
+    } catch (Exception $e){
+    }
+
+    $serverconfig = $state['privacyidea:serverconfig'];
+    $pushToken = $_GET['token'];
+    $authToken = sspmod_privacyidea_Auth_utils::fetchAuthToken($serverconfig);
+
+    $result = false;
+
+    foreach (
+        sspmod_privacyidea_Auth_utils::curl(
+            array(),
+            array("authorization:" . $authToken),
+            $serverconfig,
+            "/token/challenges/" . $pushToken,
```

Comment:
I think we should urlencode ``$pushToken`` here, to prevent the user from passing additional parameters by sending a token like ``abc&def=ihj``

Reply:
Fixed.

Reply:
Or not. We can't urlencode the name of the token here, since the server will fail to urldecode the token. So if the token is called – for instance – `a/a`, we have to actually call `/token/challenges/a/a`, `/token/challenges/a%2Fa` will be 404. Worse jet, a token ID can legally contain a question mark. How do I need to encode this so a can fetch the challenges for a token named `foo?bar=baz`? I can't seem to fin anything in the documentation.

Reply:
Since this is fixed with PI 3.2 anyways, and only affects edge-cases anyways, we can safely ignore this. I have added a comment explaining this caveat.